core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -130,7 +142,7 @@ private[spark] class CoarseMesosSchedulerBackend(
       command.setValue(
         "%s \"%s\" org.apache.spark.executor.CoarseGrainedExecutorBackend"
           .format(prefixEnv, runScript) +
-        s" --driver-url $driverUrl" +
+        s" --driver-url $driverURL" +
         s" --executor-id ${offer.getSlaveId.getValue}" +
```

Comment:
3 Mesos slaves in the local network, driver running on my machine. I modified the code to use the task ID, as in your original PR.

I'm running spark-shell with something like this:

```
val rdd = sc.parallelize(1 to 10000000, 500)

val rdd1 = rdd.zipWithIndex.groupBy(_._1 / 100)

rdd1.cache()

rdd1.collect()
```

spark-defaults.conf:

```
spark.mesos.coarse				true
spark.shuffle.service.enabled	true

spark.dynamicAllocation.enabled				true
spark.dynamicAllocation.minExecutors 			1
spark.dynamicAllocation.maxExecutors 			3
spark.dynamicAllocation.executorIdleTimeout		15
```

I can see the executors being shut down and coming back to life...